core/system/src/mpu/vmpu_freescale_k64_mem.c

```diff
@@ -81,24 +143,33 @@ void vmpu_mem_switch(uint8_t src_box, uint8_t dst_box)
     rgd = box->acl;
     assert(rgd);
 
-    /* update MPU regions */
-    t = g_mem_acl_user + 1;
-    u = dst_count;
-    while(u--)
-    {
-        word = MPU->WORD[t++];
+    /* Update MPU regions:
+     * We assume that the first two ACLs belonging to the box are the stack
+     * and context ACLs. Therefore we start at index K64F_MPU_REGIONS_RESERVED.
+     * We opportunistically copy all remaining ACLs into the MPU regions, until
```

Comment:
@niklas-arm : please add a Fixme here to copy regions more smart using the page fault counter.